balancer/rls/internal/keys/builder.go

```diff
@@ -120,6 +120,10 @@ type KeyMap struct {
 // RLSKey builds the RLS keys to be used for the given request, identified by
 // the request path and the request headers stored in metadata.
 func (bm BuilderMap) RLSKey(md metadata.MD, host, path string) KeyMap {
+	// The path passed in is of the form "/service/method". The keyBuilderMap is
+	// indexed with keys of the form "/service/" or "/service/method". The service
+	// that we set in the keyMap (to be sent out in the RLS request) should not
+	// include any slashes though.
 	i := strings.LastIndex(path, "/")
 	service, method := path[:i+1], path[i+1:]
```

Comment:
Theoretically this could panic if the path is empty.  I think this PR can be merged as-is and we add validation of the path in another PR, OR, if you want to revert the last commit, add that validation in this PR instead.  Either way is fine with me.

Reply:
I added validation at the top of `Pick`. I left the last commit as-is as I liked using `strings.Trim` to remove both the leading and trailing slashes from `service` in line 142.

Reply:
👍